src/threads.cpp

```diff
@@ -44,6 +44,22 @@
 
 extern Config_Options options;
 
+inline void _safe_to_cancel() {
+	// see https://blog.memzero.de/pthread-cancel-noexcept/
+	pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, NULL);
+	pthread_testcancel();
+	pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, NULL);
+}
+
+inline void _cancellable_sleep(int seconds) {
+	// TODO: This could be improved by using pthread_cond_timedwait instead.
+	_safe_to_cancel();
```

Comment:
sleep is according to man 7 pthread a cancellation point. So using deferred cancellation type would fit better here.

Reply:
Revert back to normal sleep, enabled cancellation around it.